### PR TITLE
source: add accessor for the 'main' source of a project.

### PR DESCRIPTION
**Pull request by simbuerg**
This adds support for the 'main' source of a project. By default
this should be the first project listed in a project.

This holds for all existing benchbuild project's and should hold for all
known sources in other projects.

Example usage:
  >>> main(*[1,2,3])
  1

### CODE REVIEW

**Comment by simbuerg** [comment]
Should be pretty simple: Drop in all sources, get the 'main' one out. No tagging needed, yet.
The current implementation assumes that the first source is the main source. If we need a more complicated 'tagging' mechanism, we can add it later.

By definition, all sources that are not 'main' are subordinate, no tag needed either.

**Comment by simbuerg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/polyjit/benchbuild_2/pullRequest?prid=5845079&bid=19137385)

File: benchbuild/project.py [hunk 1]
```diff
@@ -227,6 +227,12 @@ def __check_if_uuid(self, _: tp.Any, value: uuid.UUID) -> None:  # pylint: disab
     source: Sources = attr.ib(
         default=attr.Factory(lambda self: type(self).SOURCE, takes_self=True))
 
+    primary_source: source.BaseSource = attr.ib(init=False)
+
+    @primary_source.default
+    def __default_primary_source(self) -> str:  # pylint: disable=no-self-use
+        return source.primary(*self.source).key
```
Comment (boehmseb) [inline benchbuild/project.py:234]: Python complains to me that an instance of my project class (that inherits from `Project`) has no member with the name `source`. Not sure why, but using the `SOURCE` class attribute works.
I also just encountered a similar behavior with `Experiment.name`. Maybe it is a problem related to `attr`?

Comment (simbuerg) [reply benchbuild/project.py:234]: hmmm not sure either, I experience weirdness with TypeErrors in the builder. Normally you can exploit
varargs to demand at least 1 function argument like I did with primary(...). Works fine in a python console  too, but yields TypeErrors during testing.

Comment (simbuerg) [reply benchbuild/project.py:234]: Ah My Bad. That happends when the TestProject does not declare any source :-) Works as intended heh

Comment (boehmseb) [reply benchbuild/project.py:234]: Seems to work now. Was the `init=False` the cause?

Comment (boehmseb) [reply benchbuild/project.py:234]: Seems to work now. Was the `init=False` the cause?

Comment (simbuerg) [reply benchbuild/project.py:234]: Not sure why it should change anything. Was a guess from my side. The attribute should be defined as soon as python loads the class.

**Pull request merged**